Classes/Event/EventTypeService.php

```diff
@@ -0,0 +1,137 @@
+<?php
+namespace Neos\Cqrs\Event;
+
+/*
+ * This file is part of the Neos.Cqrs package.
+ *
+ * (c) Contributors of the Neos Project - www.neos.io
+ *
+ * This package is Open Source Software. For the full copyright and license
+ * information, please view the LICENSE file which was distributed with this
+ * source code.
+ */
+
+use Neos\Cqrs\Exception;
+use TYPO3\Flow\Annotations as Flow;
+use TYPO3\Flow\Object\ObjectManagerInterface;
+use TYPO3\Flow\Reflection\ReflectionService;
+use TYPO3\Flow\Utility\TypeHandling;
+
+/**
+ * EventTypeService
+ *
+ * @Flow\Scope("singleton")
+ */
+class EventTypeService
+{
+    /**
+     * @var ObjectManagerInterface
+     * @Flow\Inject
+     */
+    protected $objectManager;
+
+    /**
+     * @var array
+     */
+    protected $mapping = [];
+
+    /**
+     * @var array
+     */
+    protected $reversedMapping = [];
+
+    /**
+     * Register event listeners based on annotations
+     */
+    public function initializeObject()
+    {
+        $this->mapping = self::eventTypeMapping($this->objectManager);
+        $this->reversedMapping = array_flip($this->mapping);
+    }
+
+    /**
+     * Return the event type for the given Event object
+     *
+     * @param EventInterface $event
+     * @return string
+     */
+    public function getEventType(EventInterface $event): string
+    {
+        $classname = TypeHandling::getTypeForValue($event);
+        return $this->mapping[$classname];
+    }
+
+    /**
+     * Return the event type for the given Event classname
+     *
+     * @param string $classname
+     * @return string
+     */
+    public function getEventTypeByImplementation(string $classname): string
+    {
+        return $this->mapping[$classname];
+    }
+
+    /**
+     * Return the event short name for the given Event object
+     *
+     * @param EventInterface $event
+     * @return string
+     */
+    public function getEventShortType(EventInterface $event): string
```

Comment:
yes, but the API should be decoupled from the "implementation" if at some point we allow to replace this service, by implementing an interface, I prefer to use Event Type and Event Short Type, to let the implementation free of "technical" coupling